Code/Tools/AssetProcessor/native/ui/ProductAssetDetailsPanel.cpp

```diff
@@ -46,6 +47,42 @@ namespace AssetProcessor
 
     }
 
+    void ProductAssetDetailsPanel::SetupDependencyGraph(QTreeView* productAssetsTreeView, AZStd::shared_ptr<AssetDatabaseConnection> assetDatabaseConnection)
+    {
+        m_dependencyTreeDelegate = new ProductDependencyTreeDelegate(nullptr, this);
```

Comment:
this has no parent, are you deleting this?

Reply:
I have separated them into two delegates and make the ProductAssetDetailsPanel as the owner (by passing in `this` as delegates' parent)